src/main/java/se/liu/ida/hefquin/engine/queryplan/logical/impl/LogicalOpRequest.java

```diff
@@ -44,4 +55,36 @@ public void visit( final LogicalPlanVisitor visitor ) {
 		visitor.visit(this);
 	}
 
+	@Override
+	public String toString(){
+		final StringBuilder builder = new StringBuilder();
+
+		builder.append(" > req ");
+		builder.append("( ");
+
+		builder.append("{ ");
+		if ( req instanceof SPARQLRequest) {
+			final SPARQLGraphPattern pattern = ((SPARQLRequest) req).getQueryPattern();
+			if ( pattern instanceof TriplePattern) {
+				builder.append( ((TriplePattern)pattern).asJenaTriple() );
+			}
+			else if ( pattern instanceof BGP) {
+				builder.append( ((BGP) pattern).getTriplePatterns() );
+			}
+			else {
+				LogicalPlanUtils.printTriplesOfGraphPattern( pattern.asJenaOp(), builder );
+			}
+		}
+		else {
+			throw new UnsupportedOperationException( "Print graph pattern of the Request: " + req.getClass().getName()+" is an open TODO." );
+		}
```

Comment:
Perhaps we can simply rely on the `toString()` method of the classes that implement `DataRetrievalRequest`?